tests/Runner/FileFilterIteratorTest.php

```diff
@@ -0,0 +1,188 @@
+            new \ArrayIterator(array_fill(0, $repeat, $fileInfo)),
+            $eventDispatcher,
+            $cache->reveal()
+        );
+
+        $this->assertCount(0, $events);
+
+        $files = iterator_to_array($filter);
+
+        $this->assertCount(1, $files);
+        $this->assertSame($fileInfo, reset($files));
+    }
+
+    public function testEmitSkipEventWhenCacheNeedFixingFalse()
+    {
+        $file = __FILE__;
+        $content = file_get_contents($file);
+        $events = array();
+
+        $eventDispatcher = new EventDispatcher();
+        $eventDispatcher->addListener(
+            FixerFileProcessedEvent::NAME,
+            static function ($event) use (&$events) {
+                $events[] = $event;
+            }
+        );
+
+        $cache = $this->prophesize(CacheManagerInterface::class);
+        $cache->needFixing($file, $content)->willReturn(false);
+
+        $filter = new FileFilterIterator(
+            new \ArrayIterator(array(new \SplFileInfo($file))),
+            $eventDispatcher,
+            $cache->reveal()
+        );
+
+        $this->assertCount(0, $filter);
+        $this->assertCount(1, $events);
+
+        /** @var FixerFileProcessedEvent $event */
+        $event = reset($events);
+
+        $this->assertInstanceOf(FixerFileProcessedEvent::class, $event);
+        $this->assertSame(FixerFileProcessedEvent::STATUS_SKIPPED, $event->getStatus());
+    }
+
+    public function testIgnoreEmptyFile()
+    {
+        $file = __DIR__.'/../Fixtures/empty.php';
+        $content = file_get_contents($file);
+        $events = array();
+
+        $eventDispatcher = new EventDispatcher();
+        $eventDispatcher->addListener(
+            FixerFileProcessedEvent::NAME,
+            static function ($event) use (&$events) {
+                $events[] = $event;
+            }
+        );
+
+        $cache = $this->prophesize(CacheManagerInterface::class);
+        $cache->needFixing($file, $content)->willReturn(true);
+
+        $filter = new FileFilterIterator(
+            new \ArrayIterator(array(new \SplFileInfo($file))),
+            $eventDispatcher,
+            $cache->reveal()
+        );
+
+        $this->assertCount(0, $filter);
+        $this->assertCount(1, $events);
+
+        /** @var FixerFileProcessedEvent $event */
+        $event = reset($events);
+
+        $this->assertInstanceOf(FixerFileProcessedEvent::class, $event);
+        $this->assertSame(FixerFileProcessedEvent::STATUS_SKIPPED, $event->getStatus());
+    }
+
+    public function testIgnore()
+    {
+        $eventDispatcher = new EventDispatcher();
+        $eventDispatcher->addListener(
+            FixerFileProcessedEvent::NAME,
+            function () {
+                throw new \Exception('No event expected.');
+            }
+        );
+
+        $filter = new FileFilterIterator(
+            new \ArrayIterator(array(
+                new \SplFileInfo(__DIR__),
+                new \SplFileInfo('__INVALID__'),
```

Comment:
not sure about these type of invalid `\SplFileInfo` objects, should we try to detect and throw an exception or silent ignore?

Reply:
I wouldn't go that far, it will fail anyway on reading the content, right ?

Reply:
for now we would skip the entry because it is not a file, we used to crash on these